modules/services/imagery_offset.js

```diff
@@ -0,0 +1,165 @@
+import * as d3 from 'd3';
+import _ from 'lodash';
+import rbush from 'rbush';
+import { geoExtent } from '../geo/index';
+import { utilQsString } from '../util/index';
+
+var apibase = 'http://offsets.textual.ru/get?',
+    inflight = {},
+    offsetCache;
+
+export default {
+    init: function() {
+        inflight = {};
+        offsetCache = rbush();
+        window.offsetCache = offsetCache;
+    },
+
+    reset: function() {
+        _.forEach(inflight, function(req) {
+            req.abort();
+        });
+        inflight = {};
+        offsetCache = rbush();
+    },
+    getImageryID: function(url, imageryType) {
```

Comment:
@Zverik could I get your review for this piece of code, I tried to convert the Java [code](https://trac.openstreetmap.org/browser/subversion/applications/editors/josm/plugins/imagery_offset_db/src/iodb/ImageryIdGenerator.java) you wrote in Javascript.

If you could also point me to some unit test cases for this function in your code base ..